### PR TITLE
feat(Button): button component.

### PR DESCRIPTION
**Pull request by lastDunadan**
Working Button component, very universal. Can be used as 'submit', 'onclick' and 'anchor'.

### CODE REVIEW

**Comment by kubaf13** [comment]
Dlaczego nie używasz 'theme'?

File: src/components/Button/Button.styled.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { styled, css } from 'styled-components'
+import { COLOR_VARIANTS, FONT_VARIANTS } from './Button.types';
+
+const fontBasic = css`
+  font-family: Lato, sans-serif;
```
Comment (kubaf13) [inline src/components/Button/Button.styled.ts:5]: GlobalStyles add `Lato` as default font

File: src/components/Button/Button.types.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+export interface ButtonProps {
+  label: string,
+  colorVariant: COLOR_VARIANTS,
+  fontVariant: FONT_VARIANTS,
+  purpose: PURPOSE,
+  anchorLink?: string,
+  onclickAction?: () => void,
+}
+
+export type PurposeSelector = (
```
Comment (kubaf13) [inline src/components/Button/Button.types.ts:26]: `PurposeSelectorProps`

File: src/components/Button/index.tsx [hunk 3]
```diff
@@ -0,0 +1 @@
+export { Button as default } from './Button';
```
Comment (kubaf13) [inline src/components/Button/index.tsx:1]: export type { ButtonProps,  PurposeSelectorProps } from './Button.types';

File: src/components/Button/Button.styled.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  font-size: 18px;
+`
+
+export const ButtonWrapper = styled.div<{
+  bgcolorvariant: COLOR_VARIANTS,
+  fontcapvariant: FONT_VARIANTS
+}>`
+  button {
+    text-transform: ${({fontcapvariant}) => fontcapvariant === FONT_VARIANTS.STANDARD ? 'none' : 'uppercase'};
+    color: ${({bgcolorvariant}) => bgcolorvariant === COLOR_VARIANTS.GREEN ? '#ffffff' : '#000000'};
```
Comment (kubaf13) [inline src/components/Button/Button.styled.ts:15]: use `theme`

File: src/components/Button/Button.styled.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    background: ${({bgcolorvariant}) => bgcolorvariant === COLOR_VARIANTS.GREEN ? '#6C7716' : '#fefefe'};
+  
+    &:active {
+      background: ${({bgcolorvariant}) => bgcolorvariant === COLOR_VARIANTS.GREEN ? '#7E8B1A' : '#eeeeee'};
```
Comment (kubaf13) [inline src/components/Button/Button.styled.ts:19]: use theme

File: src/components/Button/Button.styled.ts (hunk 1 above)
Comment (kubaf13) [reply src/components/Button/Button.styled.ts:5]: i think you can add this styles to `GlobalStyle` file

File: src/components/Button/Button.styled.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    }
+  }
+`;
+
+export const Anchor = styled.a`
+  ${fontBasic};
+`;
+
+export const ButtonBody = styled.button`
```
Comment (kubaf13) [inline src/components/Button/Button.styled.ts:28]: I think all buttons in DEMO will look like this so in my opinion these props should be in GlobalStyles file

**Review by kubaf13 — CHANGES_REQUESTED** [review]
need to change

File: src/components/Button/Button.styled.ts (hunk 1 above)
Comment (lastDunadan) [reply src/components/Button/Button.styled.ts:5]: Instruction moved to `GlobalStyles` like so:

```css
button {
  font-family: ${({ theme }) => theme.fonts.lato};
  font-size: 18px;
}
```

File: src/components/Button/Button.styled.ts (hunk 6 above)
Comment (lastDunadan) [reply src/components/Button/Button.styled.ts:28]: But all buttons also will be this component so I'd leave them in component folder.

File: src/components/Button/Button.styled.ts (hunk 4 above)
Comment (lastDunadan) [reply src/components/Button/Button.styled.ts:15]: done

File: src/components/Button/Button.styled.ts (hunk 5 above)
Comment (lastDunadan) [reply src/components/Button/Button.styled.ts:19]: done

File: src/components/Button/Button.types.ts (hunk 2 above)
Comment (lastDunadan) [reply src/components/Button/Button.types.ts:26]: That is a **function type, not argumets/props type**. I do not think I've done function types in a way You sugessted.

File: src/components/Button/index.tsx (hunk 3 above)
Comment (lastDunadan) [reply src/components/Button/index.tsx:1]: done

**Review by kubaf13 — APPROVED** [review]

**Review by kubaf13 — APPROVED** [review]

**Pull request merged**